### PR TITLE
Update HTML purifier

### PR DESCRIPTION
**Pull request by MathiasReker**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Update to latest version of HTMLPurifier v. 4.10.0. Since old version some bugs are solved and it is compatible with PHP 7.2.
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/10128 and https://github.com/PrestaShop/PrestaShop/issues/9685
| How to test?  | Test HTML Purifier by running `composer install`, and modify the content of a text field with TinyMce

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/10376)

### CODE REVIEW

**Comment by PierreRambaud** [comment]
Yes!! Thanks @Mreker!
@ttoine can you check the license please?
@mickaelandrieu @Quetzacoalt91 if you know how HTML purifier work :)

File: tools/htmlpurifier/HTMLPurifier.standalone.php [hunk 1]
```diff
@@ -332,8 +333,8 @@ public static function arborize($tokens, $config, $context) {
             if ($token instanceof HTMLPurifier_Token_End) {
                 $token->start = null; // [MUT]
                 $r = array_pop($stack);
-                assert($r->name === $token->name);
-                assert(empty($token->attr));
+                //assert($r->name === $token->name);
```
Comment (mickaelandrieu) [inline tools/htmlpurifier/HTMLPurifier.standalone.php]: I'm worried about commented code: we use the real version of 4.10 or we don't use it but there is no way we "hack" it. Also why not use Composer to do that instead? https://packagist.org/packages/ezyang/htmlpurifier

**Comment by Quetzacoalt91** [comment]
Hi,

That's a open discussion I start here, but I see that HTML purifier is available through Composer (http://htmlpurifier.org/download#Composer).
That is probably a nice moment to remove it from our files and integrate it in our composer file?

File: tools/htmlpurifier/HTMLPurifier.standalone.php (hunk 1 above)
Comment (MathiasReker) [reply tools/htmlpurifier/HTMLPurifier.standalone.php]: The code is commented by HTMLPurifier in the original 4.10.0.

Comment (mickaelandrieu) [reply tools/htmlpurifier/HTMLPurifier.standalone.php]: Sad !

giphy

**Comment by MathiasReker** [comment]
Fine with me, but I am not able to do it. I am not that skilled sorry.

File: tools/htmlpurifier/HTMLPurifier.standalone.php (hunk 1 above)
Comment (MathiasReker) [reply tools/htmlpurifier/HTMLPurifier.standalone.php]: Do you want me to delete commented lines?

**Comment by MathiasReker** [comment]
Maybe it can wait? Now the HTMLPurifier is at least up to date.

File: tools/htmlpurifier/HTMLPurifier.standalone.php (hunk 1 above)
Comment (mickaelandrieu) [reply tools/htmlpurifier/HTMLPurifier.standalone.php]: no, it's not your fault, it's HTMLPurifier's to do it ^^

**Comment by mickaelandrieu** [comment]
WDYT @Quetzacoalt91? Should we merge this first, or do you have time to do the pull request using Composer?

**Comment by Quetzacoalt91** [comment]
I'll create another branch from your changes @Mreker and I let you know.

**Comment by Quetzacoalt91** [comment]
Done, HtmlPurifier is now included with Composer.

**Review by mickaelandrieu — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by mbadrani** [comment]
Seems ok for me, sending formular request and compliant with PHP 7.0.32.
@Quetzacoalt91 @eternoendless if anyone have a PHP 7.2 to confirm that is also compliant

**Comment by Quetzacoalt91** [comment]
Thank you everybody for review, and @Mreker for having initiated this task.

**Pull request merged**